src/ds/d500/d500-auto-calibration-handler.cpp

```diff
@@ -0,0 +1,125 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
+
+#include "d500-auto-calibration-handler.h"
+
+
+namespace librealsense
+{
+    d500_calibration_answer d500_auto_calibrated_handler_hw_monitor::get_status() const
+    {
+        if (auto hwm = _hw_monitor.lock())
+        {
+            auto res = hwm->send(command{ ds::GET_CALIB_STATUS });
+            // checking size of received buffer
+            if (!check_buffer_size_from_get_calib_status(res))
+                throw std::runtime_error("GET_CALIB_STATUS returned struct with wrong size");
+
+            return *reinterpret_cast<d500_calibration_answer*>(res.data());
+        }
+        throw std::runtime_error("hw monitor has not been set");
+    }
+
+    std::vector<uint8_t> d500_auto_calibrated_handler_hw_monitor::run_auto_calibration(d500_calibration_mode _mode)
+    {
+        if (auto hwm = _hw_monitor.lock())
+        {
+            return hwm->send(command{ ds::SET_CALIB_MODE, static_cast<uint32_t>(_mode), 1 /*always*/ });
+        }
+        throw std::runtime_error("hw monitor has not been set");
+    }
+
+    void d500_auto_calibrated_handler_hw_monitor::set_hw_monitor_for_auto_calib(std::shared_ptr<hw_monitor> hwm)
+    {
+        _hw_monitor = hwm;
+    }
+
+    bool d500_auto_calibrated_handler_hw_monitor::check_buffer_size_from_get_calib_status(std::vector<uint8_t> res) const
+    {
+        // the GET_CALIB_STATUS command will return:
+        // - 3 bytes during the whole process
+        // - 515 bytes (3 bytes + 512 bytes of the depth calibration) when the state is Complete
+
+        bool is_size_ok = false;
+        if (res.size() > 1)
+        {
+            if (res[0] < static_cast<int>(d500_calibration_state::RS2_D500_CALIBRATION_STATE_COMPLETE) &&
+                res.size() == (sizeof(d500_calibration_answer) - sizeof(ds::d500_coefficients_table)))
+                is_size_ok = true;
+
+            if (res[0] == static_cast<int>(d500_calibration_state::RS2_D500_CALIBRATION_STATE_COMPLETE) &&
```

Comment:
Why is this splitted into 2 condition that check the same thing with a small diff and override the is_size_ok?

merge them?